File: 1895-minimum-number-of-operations-to-move-all-balls-to-each-box/minimum-number-of-operations-to-move-all-balls-to-each-box.py
class Solution:
    def minOperations(self, boxes: str) -> List[int]:
        output = [0] * len(boxes)
        moves = 0
        balls = 0

        for i in range(len(boxes)):
            output[i] = moves + balls
            moves += balls
            if (boxes[i] == "1"):
                balls += 1

        moves = 0
        balls = 0
        print(output)
        for i in range(len(boxes) - 1, -1, -1):
            output[i] += moves + balls

            moves += balls
            if (boxes[i] == "1"):
                balls += 1

        return output
        

File: 1895-minimum-number-of-operations-to-move-all-balls-to-each-box/minimum-number-of-operations-to-move-all-balls-to-each-box.js
var minOperations = function(boxes) {
    let output = new Array(boxes.length).fill(0)
    let moves = 0
    let balls = 0

    for (let i = 0; i < boxes.length; i++) {
        let ball = parseInt(boxes[i])
        output[i] = balls + moves
        moves += balls
        balls += ball
    }    

    balls = 0
    moves = 0
    for (let i = boxes.length - 1; i >= 0; i--) {
        let ball = parseInt(boxes[i])
        output[i] += balls + moves
        moves += balls
        balls += ball
    }

    return output
};